pkg/cmd/kafka/create/create.go

```diff
@@ -174,48 +180,70 @@ func runCreate(opts *Options) error {
 		}
 	}
 
-	logger.Info(opts.localizer.MustLocalize("kafka.create.log.info.creatingKafka", localize.NewEntry("Name", payload.Name)))
-
 	api := connection.API()
 
 	a := api.Kafka().CreateKafka(context.Background())
 	a = a.KafkaRequestPayload(*payload)
 	a = a.Async(true)
 	response, httpRes, err := a.Execute()
 
-	if httpRes.StatusCode == 409 {
+	if httpRes.StatusCode == http.StatusBadRequest {
 		return errors.New(opts.localizer.MustLocalize("kafka.create.error.conflictError", localize.NewEntry("Name", payload.Name)))
 	}
 
 	if err != nil {
 		return err
 	}
 
-	logger.Info(opts.localizer.MustLocalize("kafka.create.info.successMessage", localize.NewEntry("Name", response.GetName())))
-
-	switch opts.outputFormat {
-	case dump.JSONFormat:
-		data, _ := json.MarshalIndent(response, "", cmdutil.DefaultJSONIndent)
-		_ = dump.JSON(opts.IO.Out, data)
-	case dump.YAMLFormat, dump.YMLFormat:
-		data, _ := yaml.Marshal(response)
-		_ = dump.YAML(opts.IO.Out, data)
-	}
-
 	kafkaCfg := &config.KafkaConfig{
 		ClusterID: response.GetId(),
 	}
 
 	if opts.autoUse {
 		logger.Debug("Auto-use is set, updating the current instance")
 		cfg.Services.Kafka = kafkaCfg
-		if err := opts.Config.Save(cfg); err != nil {
+		if err = opts.Config.Save(cfg); err != nil {
 			return fmt.Errorf("%v: %w", opts.localizer.MustLocalize("kafka.common.error.couldNotUseKafka"), err)
 		}
 	} else {
 		logger.Debug("Auto-use is not set, skipping updating the current instance")
 	}
 
+	if opts.wait {
+		logger.Debug("--wait flag is enabled, waiting for Kafka to finish creating")
+		s := opts.IO.NewSpinner()
+		s.Suffix = fmt.Sprintf(" %v", opts.localizer.MustLocalize("kafka.create.log.info.creatingKafka", localize.NewEntry("Name", opts.name)))
+		s.Start()
+
+		for svcstatus.IsCreating(response.GetStatus()) {
+			time.Sleep(cmdutil.DefaultPollTime)
+			s.Suffix = " " + opts.localizer.MustLocalize("kafka.create.log.info.creationInProgress", localize.NewEntry("Name", response.GetName()), localize.NewEntry("Status", color.Info(response.GetStatus())))
+			response, httpRes, err = api.Kafka().GetKafkaById(context.Background(), response.GetId()).Execute()
+			logger.Debug("Checking Kafka status:", response.GetStatus())
+			if err != nil {
+				return err
+			}
+		}
+		s.Stop()
+		logger.Info()
+		logger.Info()
+		logger.Info(opts.localizer.MustLocalize("kafka.create.info.successSync", localize.NewEntry("Name", response.GetName())))
+	}
+
+	switch opts.outputFormat {
```

Comment:
FYI There is helper for that in dump package. We no longer need this boilerplate code

Reply:
Yeah, I'm aware. Would like that done another time (this switch already existed).

Reply:
Sorry for offtopic.